File: src/main/java/com/refinedmods/refinedstorage/screen/grid/filtering/TagGridFilter.java
package com.refinedmods.refinedstorage.screen.grid.filtering;

import com.refinedmods.refinedstorage.screen.grid.stack.IGridStack;

import java.util.function.Predicate;

public class TagGridFilter implements Predicate<IGridStack> {
    private final String tagName;

    public TagGridFilter(String tagName) {
        this.tagName = tagName.toLowerCase();
    }

    @Override
    public boolean test(IGridStack stack) {
        return stack.getTags().stream().anyMatch(name -> name.toLowerCase().contains(this.tagName));
    }
}

File: src/main/kotlin/com/refinedmods/refinedstorage/screen/grid/filtering/TagGridFilter.kt
package com.refinedmods.refinedstorage.screen.grid.filtering

import com.refinedmods.refinedstorage.screen.grid.stack.IGridStack
import java.util.function.Predicate

class TagGridFilter(tagName: String) : Predicate<IGridStack> {
    private val tagName: String
    override fun test(stack: IGridStack): Boolean {
        return stack.tags.stream().anyMatch { name: String? -> name!!.toLowerCase().contains(tagName) }
    }

    init {
        this.tagName = tagName.toLowerCase()
    }
}